app/logic/events.py

```diff
@@ -350,24 +353,27 @@ def validateNewEventData(data):
 
     if data['timeEnd'] <= data['timeStart']:
         return (False, "Event end time must be after start time.")
-
+    
     # Validation if we are inserting a new event
     if 'id' not in data:
 
-        event = (Event.select()
+        sameEventList = list((Event.select()
                       .where((Event.name == data['name']) &
                              (Event.location == data['location']) &
                              (Event.startDate == data['startDate']) &
-                             (Event.timeStart == data['timeStart'])))
+                             (Event.timeStart == data['timeStart'])).execute()))
+        
+        for item in sameEventList:
+            if item.isCanceled or item.recurringId:
+                sameEventList.remove(item)
```

Comment:
It is really nitpicky but I think it  would read a lot better to use `event` instead of `item` for the `for` loop variable.

Reply:
Done.